### PR TITLE
Feature/352 view templates

### LINKED ISSUE #352: Templates UI - view templates
UI implementation for templates. 
The backend has been implemented in #295 #298 

User should be able to list templates and should be able to see the template configuration.

### PR DESCRIPTION
**Pull request by jozefbakus**
Implemented the first part of #352

Implemented: 

- Job templates search endpoint - pagination, filters, sorts
- UI - Admin/Job Template page with Job Template list + pagination, filters, sorts

### CODE REVIEW

File: ui/src/app/components/admin/job-templates/job-templates-home/job-templates-home.component.html [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  <clr-dg-column [clrDgField]="jobTemplateColumns.NAME" [clrFilterValue]>
+    <ng-container *clrDgHideableColumn="{hidden: false}">Job template name</ng-container>
+    <clr-dg-filter [clrDgFilter]="templateNameFilter">
+      <app-string-filter #templateNameFilter [property]="jobTemplateColumns.NAME"
+                         [removeFiltersSubject]="removeFiltersSubject">
+      </app-string-filter>
+    </clr-dg-filter>
+  </clr-dg-column>
+  <clr-dg-column [clrDgField]="jobTemplateColumns.FROM_CONFIG">
+    <ng-container *clrDgHideableColumn="{hidden: false}">From config</ng-container>
```
Comment (kevinwallimann) [inline ui/src/app/components/admin/job-templates/job-templates-home/job-templates-home.component.html]: Typo `From`

File: ui/src/app/constants/jobTemplateColumns.constants.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export const jobTemplateColumns = {
+  NAME: 'name',
+  FROM_CONFIG: 'formConfig',
```
Comment (kevinwallimann) [inline ui/src/app/constants/jobTemplateColumns.constants.ts]: Typo `FROM`

File: ui/src/app/app.component.spec.ts [hunk 3]
```diff
@@ -81,4 +89,24 @@ describe('AppComponent', () => {
     expect(compiled.querySelector('.title')).toBeFalsy();
     expect(compiled.querySelector('.header-actions')).toBeFalsy();
   });
+
+  it('isActive(base) should return true if current rout contains base', () => {
```
Comment (kevinwallimann) [inline ui/src/app/app.component.spec.ts]: typo `rout`

File: ui/src/app/models/jobTemplate.model.ts [hunk 4]
```diff
@@ -17,10 +17,11 @@ export type JobTemplateModel = {
   id: number;
   name: string;
   formConfig: string;
+  jobType: { name: string };
 };
 
 export class JobTemplateModelFactory {
-  static create(id: number, name: string, formConfig: string): JobTemplateModel {
-    return { id: id, name: name, formConfig: formConfig };
+  static create(id: number, name: string, formConfig: string, jobType: { name: string }): JobTemplateModel {
```
Comment (kevinwallimann) [inline ui/src/app/models/jobTemplate.model.ts]: There is `JobType` in `app/models/jobInstance.model.ts`

File: ui/src/app/stores/job-templates/job-templates.reducers.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  page: 1,
+  loading: false,
+};
+
+export function jobTemplatesReducer(state: State = initialState, action: JobTemplatesActions.JobTemplatesActions) {
+  switch (action.type) {
+    case JobTemplatesActions.SEARCH_JOB_TEMPLATES:
+      return { ...state, loading: true };
+    case JobTemplatesActions.SEARCH_JOB_TEMPLATES_SUCCESS:
+      console.log('payload: ', action.payload);
```
Comment (kevinwallimann) [inline ui/src/app/stores/job-templates/job-templates.reducers.ts]: Did you forget to remove this `console.log`?

File: ui/src/app/stores/job-templates/job-templates.effects.spec.ts [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+      spyOn(jobTemplateService, 'searchJobTemplates').and.returnValue(searchJobTemplatesResponse);
+
+      expect(underTest.jobTemplatesSearch).toBeObservable(expected);
+    });
+
+    it('should return search job templates failure if jobTemplateService.searchJobTemplates responds with an error', () => {
+      const action = new SearchJobTemplates({ from: 0, size: 0, sort: new SortAttributesModel('', 0) });
+      mockActions = cold('-a', { a: action });
+      const errorResponse = cold('-#|');
+      spyOn(jobTemplateService, 'searchJobTemplates').and.returnValue(errorResponse);
```
Comment (kevinwallimann) [inline ui/src/app/stores/job-templates/job-templates.effects.spec.ts]: Btw have a look at https://github.com/AbsaOSS/hyperdrive-trigger/pull/347 (type-safe spies)

File: ui/src/app/app.component.spec.ts [hunk 7]
```diff
@@ -52,6 +57,9 @@ describe('AppComponent', () => {
     }).compileComponents();
 
     mockStore = TestBed.inject(MockStore);
+    mockRouter = TestBed.inject(Router);
+    fixture = TestBed.createComponent(AppComponent);
```
Comment (kevinwallimann) [inline ui/src/app/app.component.spec.ts:61]: Sonar: Variables should not be shadowed
`fixture` is already used in some test cases and shadow this global `fixture`

File: ui/src/app/stores/job-templates/job-templates.reducers.ts [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+const initialState: State = {
+  jobTemplates: [],
+  total: 0,
+  page: 1,
+  loading: false,
+};
+
+export function jobTemplatesReducer(state: State = initialState, action: JobTemplatesActions.JobTemplatesActions) {
```
Comment (kevinwallimann) [inline ui/src/app/stores/job-templates/job-templates.reducers.ts:33]: Sonar: Default parameters should be last

File: ui/src/app/components/admin/job-templates/job-templates-home/job-templates-home.component.html [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+    <clr-dg-filter [clrDgFilter]="jobTypeFilter">
+      <app-string-filter #jobTypeFilter [property]="jobTemplateColumns.JOB_TYPE"
+                         [removeFiltersSubject]="removeFiltersSubject">
+      </app-string-filter>
+    </clr-dg-filter>
+  </clr-dg-column>
+  <clr-dg-placeholder>We couldn't find any job templates!</clr-dg-placeholder>
+
+  <clr-dg-row *ngFor="let jobTemplate of jobTemplates" [clrDgItem]="jobTemplate">
+<!--    <clr-dg-action-overflow>-->
```
Comment (kevinwallimann) [inline ui/src/app/components/admin/job-templates/job-templates-home/job-templates-home.component.html]: I know you will use this, but in general, I would not commit commented code.

File: ui/src/app/components/admin/job-templates/job-templates-home/job-templates-home.component.html (hunk 1 above)
Comment (jozefbakus) [reply ui/src/app/components/admin/job-templates/job-templates-home/job-templates-home.component.html]: Fixed

File: ui/src/app/constants/jobTemplateColumns.constants.ts (hunk 2 above)
Comment (jozefbakus) [reply ui/src/app/constants/jobTemplateColumns.constants.ts]: Fixed

File: ui/src/app/app.component.spec.ts (hunk 3 above)
Comment (jozefbakus) [reply ui/src/app/app.component.spec.ts]: FixedFixed

File: ui/src/app/models/jobTemplate.model.ts (hunk 4 above)
Comment (jozefbakus) [reply ui/src/app/models/jobTemplate.model.ts]: Fixed, placed to new file

File: ui/src/app/stores/job-templates/job-templates.reducers.ts (hunk 5 above)
Comment (jozefbakus) [reply ui/src/app/stores/job-templates/job-templates.reducers.ts]: Fixed

File: ui/src/app/stores/job-templates/job-templates.effects.spec.ts (hunk 6 above)
Comment (jozefbakus) [reply ui/src/app/stores/job-templates/job-templates.effects.spec.ts]: Fixed

File: ui/src/app/app.component.spec.ts (hunk 7 above)
Comment (jozefbakus) [reply ui/src/app/app.component.spec.ts:61]: Fixed

File: ui/src/app/components/admin/job-templates/job-templates-home/job-templates-home.component.html (hunk 9 above)
Comment (jozefbakus) [reply ui/src/app/components/admin/job-templates/job-templates-home/job-templates-home.component.html]: Fixed

File: ui/src/app/stores/job-templates/job-templates.reducers.ts (hunk 8 above)
Comment (jozefbakus) [reply ui/src/app/stores/job-templates/job-templates.reducers.ts:33]: In this case, it is not possible. You have to respect ActionReducer<State, Action>

**Review by kevinwallimann — APPROVED** [review]

**Pull request merged**